### ISSUE TITLE
[Macos] Add support for YUV in externalTextureMetal 

### ISSUE THREAD

**Issue by linxuebin1990** [flutter/flutter#89042]
According to [FlutterExternalTextureMetal.mm](https://github.com/flutter/engine/blob/master/shell/platform/darwin/macos/framework/Source/FlutterExternalTextureMetal.mm),  the following code:
```objective-C
- (BOOL)populateTexture:(FlutterMetalExternalTexture*)textureOut {
  // ...
  CVReturn cvReturn =
      CVMetalTextureCacheCreateTextureFromImage(/*allocator=*/kCFAllocatorDefault,
                                                /*textureCache=*/_darwinMetalContext.textureCache,
                                                /*sourceImage=*/pixelBuffer,
                                                /*textureAttributes=*/nullptr,
                                                /*pixelFormat=*/MTLPixelFormatBGRA8Unorm,
                                                /*width=*/textureSize.width(),
                                                /*height=*/textureSize.height(),
                                                /*planeIndex=*/0u,
                                                /*texture=*/&cvMetalTexture);

   // ...
  textureOut->pixel_format = FlutterMetalExternalTexturePixelFormat::kRGBA;
  textureOut->textures = _textures.data();

  return YES;
}
```
We can know that currently only RGBA is supported.

**Comment by chinmaygarde** [comment]
I think this is a reasonable feature request. We already support this in the embedder API. So the implementation can can refer to that.

**Comment by doesnotexist** [comment]
I believe this issue can be closed.  Doesn't this merged PR https://github.com/flutter/engine/pull/28341 address the issue?

**Comment by linxuebin1990** [comment]
> 

Yes, I will close this issue.

**Comment by linxuebin1990** [comment]
> I believe this issue can be closed. Doesn't this merged PR [flutter/engine#28341](https://github.com/flutter/engine/pull/28341) address the issue?

Yes, I will close this issue.
